core-tests/jvm/src/test/scala/zio/stm/StmIssue1587Spec.scala

```diff
@@ -0,0 +1,65 @@
+package zio.stm
+
+import zio.{ TestRuntime, ZIO }
+import scala.util.control.Breaks._
+
+class StmIssue1587Spec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends TestRuntime {
```

Comment:
Props for this test to @dkarlinsky!

Reply:
Test is good
![Screenshot from 2019-09-08 08-48-18](https://user-images.githubusercontent.com/2694353/64484632-71fc0a00-d215-11e9-9a5a-21117586ebc7.png)